docs/cql/tombstone-gc.rst

```diff
@@ -0,0 +1,109 @@
+=====================================
+Tombstone Garbage Collection (GC) 
+=====================================
+
+Overview
+----------
+When you delete some data from the database, the data record is not immediately 
+removed. Instead, it is marked with a tombstone - a data marker that prevents 
+the data from being returned as a query result. The data covered by the tombstone 
+and the tombstone itself are eventually removed after the compaction process 
+is performed.
```

Comment:
This sentence can be misleading - it suggests that maybe the tombstone is removed after any compaction, which isn't true. Maybe it should be something like this (maybe can be further improved:

"The data covered by the tombstone will be removed when compaction is performed, but the tombstone itself might need to live even after the compaction - as long there is a risk that repair with other nodes will bring back the deleted data, we need to remember it was deleted and for this we keep the tombstones. But eventually, we want to delete, or _garbage-collect_, these tombstones, but this needs to be coordinated with the repair, as we'll explain now:"

(you also have some good points below, so this paragraph I wrote here isn't perfect).